### DISCUSSION TITLE
Edited Beatmaps with Video Results in File Corruption

### DISCUSSION THREAD

**Discussion by iSaturnx** [ppy/osu#23807] [Q&A]
### Type

Game behaviour

### Bug description

When editing an imported beatmap with a video, trips of importing and exporting between stable and lazer will corrupt the beatmap. In the 'verify' section of the editor, it displays "Problem: "" is a 0-byte file."

I've tried to replicate this but, it appears to only happen after some importing and exporting back and forth between stable and lazer, with having to add the video back in the 'design' section or 'open .osu in text editor' every time, since the video gets removed from the .osu file after saving an edit in lazer.
Also, have noticed this happens when extracting the .osz file as a .zip, then compressing it back into .osz.

The beatmap isn't entirely corrupt as it still imports into lazer just fine. However, it doesn't import into stable (automatically put in the "Failed" folder), nor does extracting it as a .zip file works. Though, I'm assuming I shouldn't be extracting it as a .zip file either way(?).

I'm unsure how to fix this as I can't access the video file nor open it in stable. Only way for me to go on is to create a new beatmap and add everything back individually (which is very time consuming since I can't copy and paste between beatmaps). 

### Screenshots or videos

<img width="1440" alt="Screenshot 2023-06-07 at 2 42 54 AM" src="https://github.com/ppy/osu/assets/68358837/42d319d6-4e76-4952-9d08-b31b2405e42b">
<img width="308" alt="Screenshot 2023-06-07 at 2 51 56 AM" src="https://github.com/ppy/osu/assets/68358837/6a8ecbb5-f353-4a36-a9c3-b59e163bae3e">

### Version

2023.605.0

### Logs

[database.log](https://github.com/ppy/osu/files/11675314/database.log)
[network.log](https://github.com/ppy/osu/files/11675315/network.log)
[performance.log](https://github.com/ppy/osu/files/11675316/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11675318/runtime.log)

**Comment by peppy** [comment]
Please include the beatmap you're seeing issues with.

And please don't manually extract / repack osz files. That's not supported behaviour.

**Comment by iSaturnx** [comment]
[beatmap.zip](https://github.com/ppy/osu/files/11683654/beatmap.zip) (this is the one shown in the images without extracting/repacking)

**Comment by peppy** [comment]
The linked beatmap is a corrupt osz file.

How did you create it?

**Comment by iSaturnx** [comment]
Through the editor in lazer. Eventually became corrupted after a couple times of importing it into stable to fix the video (by selecting it in 'design') and importing it back into lazer. I still have the last import of the beatmap into stable [here](https://github.com/ppy/osu/files/11684634/beatmap.zip) .

**Comment by peppy** [comment]
I don't know. I've never reproduced anything like this. Maybe you copied the file before the export finished.

**Reply by iSaturnx** [reply]
I don't think I have since I'm not sure how I can copy it before it finishes exporting? I click on it the notification to open the folder once its finished.
I'm aware that it's not easy to reproduce this and was questioning if I should even report this since I don't have further information on how this happens, but still did just in case if there is a solution to at least recovering the beatmap. This happened to another beatmap of mine out of the blue as well.
